task_2.py

```diff
@@ -20,164 +21,212 @@
-print(net)
-
-if os.path.exists('pretrained_alexnet.pkl'):
-    pret_net = pkl.load(open('pretrained_alexnet.pkl', 'rb'))
-else:
-    pret_net = model_zoo.load_url(
-        'https://download.pytorch.org/models/alexnet-owt-4df8aa71.pth')
-    pkl.dump(pret_net, open('pretrained_alexnet.pkl', 'wb'),
-             pkl.HIGHEST_PROTOCOL)
-own_state = net.state_dict()
-
-for name, param in pret_net.items():
-    print(name)
-    if name not in own_state:
-        continue
-    if isinstance(param, Parameter):
-        param = param.data
-    try:
-        own_state[name].copy_(param)
-        print('Copied {}'.format(name))
-    except:
-        print('Did not find {}'.format(name))
-        continue
-
-
-# Move model to GPU and set train mode
-net.load_state_dict(own_state)
-net.cuda()
-net.train()
-
-# TODO: Create optimizer for network parameters from conv2 onwards
-# (do not optimize conv1)
-
-
-
-
-
+# Set output directory
 output_dir = "./"
 if not os.path.exists(output_dir):
     os.makedirs(output_dir)
 
-# training
-train_loss = 0
-tp, tf, fg, bg = 0., 0., 0, 0
-step_cnt = 0
-re_cnt = False
-disp_interval = 10
-val_interval = 1000
 
-
-
-def test_net(model, val_loader=None, thresh=0.05):
+def test_model(model, val_loader=None, thresh=0.05):
     """
-    tests the networks and visualize the detections
-    thresh is the confidence threshold
+    Tests the networks and visualizes the detections
+    :param thresh: Confidence threshold
     """
+    with torch.no_grad():
+        for iter, data in enumerate(val_loader):
+
+            # one batch = data for one image
+            image           = data['image']
+            target          = data['label']
+            wgt             = data['wgt']
+            rois            = data['rois']
+            gt_boxes        = data['gt_boxes']
+            gt_class_list   = data['gt_classes']
 
-    for iter, data in enumerate(val_loader):
+            #TODO (Q2.3): perform forward pass, compute cls_probs
 
-        # one batch = data for one image
-        image           = data['image']
-        target          = data['label']
-        wgt             = data['wgt']
-        rois            = data['rois']
-        gt_boxes        = data['gt_boxes']
-        gt_class_list   = data['gt_classes']
 
-        #TODO: perform forward pass, compute cls_probs
+            # TODO (Q2.3): Iterate over each class (follow comments)
+            for class_num in range(20):
+                # get valid rois and cls_scores based on thresh
 
+                # use NMS to get boxes and scores
+                pass
 
-        # TODO: Iterate over each class (follow comments)
-        for class_num in range(20):            
-            # get valid rois and cls_scores based on thresh
-            
-            # use NMS to get boxes and scores
-            
+            #TODO (Q2.3): visualize bounding box predictions when required
+            #TODO (Q2.3): Calculate mAP on test set
```

Comment:
Are we planning to give detailed instructions for this in README? Or in FAQ? Also, since this was a long computation, maybe create an empty function like `compute_map` and ask them to fill it there instead?